fix(linter): Fix false positive on React in `consistent-type-imports`

DonIsaac opened the pull request:
## What This PR Does

Updates `typescript/consistent-type-imports` to ignore `React`.

This is not a "great" solution; ideally we would be reading the user's `tsconfig.json` and using info from that to selectively ignore/report unused `React` imports. Doing this would also fix remaining false positives in this rule. This workaround should work in most cases though.

I can make follow-up PR to have our lint runtime parse `tsconfig.json` files and make that data available to rules via `LintContext`. That change would potentially have large implications and lots of edge cases, so we should discuss it first.

DonIsaac commented:
* **#11171** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11171?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11171?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11171).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
this is horrible lol

camc314 commented:
### Merge activity

* **May 19, 3:34 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 reviewed (approved):
lgtm

DonIsaac committed:
fix(linter): Fix false positive on React in `consistent-type-imports` (#11171)

## What This PR Does

Updates `typescript/consistent-type-imports` to ignore `React`.

This is not a "great" solution; ideally we would be reading the user's `tsconfig.json` and using info from that to selectively ignore/report unused `React` imports. Doing this would also fix remaining false positives in this rule. This workaround should work in most cases though.

I can make follow-up PR to have our lint runtime parse `tsconfig.json` files and make that data available to rules via `LintContext`. That change would potentially have large implications and lots of edge cases, so we should discuss it first.
